mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/scaladsl/MqttSource.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming.scaladsl
+
+import akka.actor.ActorSystem
+import akka.{Done, NotUsed}
+import akka.event.{Logging, LoggingAdapter}
+import akka.stream.OverflowStrategy
+import akka.stream.alpakka.mqtt.streaming.impl.Setup
+import akka.stream.alpakka.mqtt.streaming._
+import akka.stream.scaladsl.{BroadcastHub, Flow, Keep, Source, SourceQueueWithComplete, Tcp}
+
+import scala.collection.immutable
+import scala.concurrent.{Future, Promise}
+
+object MqttSource {
+
+  def atMostOnce(settings: MqttSessionSettings,
+                 clientId: String,
+                 subscriptions: immutable.Seq[(String, ControlPacketFlags)]): Source[Publish, Future[Done]] =
+    Setup
+      .source { implicit materializer => implicit attributes =>
+        implicit val system: ActorSystem = materializer.system
+        implicit val logging: LoggingAdapter = Logging.getLogger(system, this)
+
+        val mqttClientSession = ActorMqttClientSession(settings)
+        val mqttFlow: Flow[Command[Nothing], Either[MqttCodec.DecodeError, Event[Nothing]], NotUsed] =
+          Mqtt
+            .clientSessionFlow(mqttClientSession)
+            .join(Tcp(system).outgoingConnection("localhost", 1883))
+
+        val subscribed = Promise[Done]()
```

Comment:
FYI You may get partial subscriptions ie the subscription response contains info on what was actually subscribed to.

Reply:
You mean the materialized future should expose information from the `SubAck`? The `ControlPacketFlags`? 
Or should it even expect several `SubAck`s?